### ISSUE TITLE
Plotting boolean data fails

### ISSUE THREAD

**Issue by BSchilperoort** [pandas-dev/pandas#23719]
#### Code example
```python
import pandas as pd
df = pd.DataFrame({'data': [True, False, False]})
df['data'].plot()
```
#### Problem description
Plotting boolean data returns _TypeError: Empty 'DataFrame': no numeric data to plot_. 
Matplotlib interprets boolean data as 0/1, which allows plotting.

#### Expected Output
A plot of the boolean data like matplotlib would make (interpreting True/False as 1/0).

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.4.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 78 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: en
LOCALE: None.None

pandas: 0.23.4
pytest: 3.3.2
pip: 9.0.1
setuptools: 38.4.0
Cython: 0.27.3
numpy: 1.14.2
scipy: 1.0.0
pyarrow: None
xarray: 0.10.9
IPython: 6.2.1
sphinx: 1.6.6
patsy: 0.5.0
dateutil: 2.6.1
pytz: 2017.3
blosc: None
bottleneck: 1.2.1
tables: 3.4.2
numexpr: 2.6.4
feather: None
matplotlib: 2.1.2
openpyxl: 2.4.10
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.0.2
lxml: 4.1.1
bs4: 4.6.0
html5lib: 1.0.1
sqlalchemy: 1.2.1
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

**Comment by gfyoung** [comment]
@BSchilperoort : Good catch!  Investigation and PR are welcome!

cc @TomAugspurger

**Comment by BSchilperoort** [comment]
Added the simple fix in a PR: #23813
